contrib/custom-image-kubelet/genconfig.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import io
+import logging
+import click
+import requests
+import boto3
+import tarfile
+from base64 import b64decode
+from yaml import load, dump
+from os.path import basename, normpath
+from collections import defaultdict
+from urllib.parse import urlparse
+from shutil import copyfileobj
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+
+IMAGES = ['kube-apiserver', 'kube-controller-manager', 'kube-scheduler', 'pause', 'etcd']
+COMPONENTS = ['kube-proxy', 'coredns', 'metrics-server']
+
+
+@click.command(context_settings={'max_content_width': 120})
+@click.option('--release-url', type=str, required=True, help='URL to a Kubernetes Release YAML')
+@click.option('--data-dir', type=str, default='/var/lib/rancher/rke2', help='RKE2 data directory', show_default=True)
+@click.option('--prefix', type=str, default='/', help='Prefix for output files created by this script', show_default=True)
+def main(release_url, data_dir, prefix):
+    release = get_release(release_url)
+
+    logging.info(f'Got Release: {release["metadata"]["name"]}')
+
+    rke2_config = {
+            'kubelet-path': normpath(f'{data_dir}/opt/bin/kubelet'),
+            'data-dir': normpath(data_dir),
+            }
+
+    for image in IMAGES:
+        alt_image = get_image_artifact(release, image)
+        rke2_config[f'{image}-image'] = alt_image
+
+    for component in COMPONENTS:
+        alt_image = get_image_artifact(release, component)
+        write_chart_config(component, alt_image, f'{prefix}/{data_dir}')
+
+    extract_archive(release, release_url, 'kubernetes-node-linux-amd64.tar.gz', f'{prefix}/{data_dir}')
+
+    try:
+        write_ecr_credentials(release, prefix)
+    except Exception as e:
+        logging.warning(f'Unable to write ECR credenials to registries.yaml: {e}')
```

Comment:
Should we exit at this point?

Reply:
Fixed the typo, moved earlier, and converted to an exception.